pkg/common/catalog/configure.go

```diff
@@ -177,12 +226,23 @@ func (v1 *configurerV1) Configure(ctx context.Context, coreConfig CoreConfig, hc
 	return err
 }
 
-func (v1 *configurerV1) Validate(ctx context.Context, coreConfig CoreConfig, hclConfiguration string) error {
-	_, err := v1.ConfigServiceClient.Validate(ctx, &configv1.ValidateRequest{
+func (v1 *configurerV1) Validate(ctx context.Context, coreConfig CoreConfig, hclConfiguration string) (*ValidateResult, error) {
+	response, err := v1.ConfigServiceClient.Validate(ctx, &configv1.ValidateRequest{
 		CoreConfiguration: coreConfig.v1(),
 		HclConfiguration:  hclConfiguration,
 	})
-	return err
+	result := NewValidateResult()
+	// TODO: The GRPC default behavior is to not return a *ValidateResponse when there is an error, and that destroys many
+	//       of the validation messages, except for the error message.  We may wish to not consider validation errors
+	//       as errors going forward.
```

Comment:
I personally think that an error should be returned only when there is an error condition during the validation process, that prevents the validation to finish, and not return an error in case of invalid settings.

Reply:
I share your opinion, but the GRPC generated code doesn't.  The only option that is possible is to revisit #5303 and not have any of the Validation() plugin funs return an error, ever.  I personally think that's a better solution than the other options, but it would require revisiting https://github.com/spiffe/spire-plugin-sdk/pull/55 to not simply have a list of notes and a "valid" flag.  Instead it would require a list of notes, and some indication of which notes are errors.

Reply:
@amartinezfayo I've updated the current PR to improve the validation in ways that align with your wishes (only use the error for API call errors) but the spire-plugin-sdk doesn't include any marking of which note contains the error.  To accomodate this under reporting in the spire-plugin-sdk, if the Valid flag is false, then catalog adds an additional error note that "the plugin configuration is invalid"

This is likely as good as it will get, until changes are made in spire-plugin-sdk.

Reply:
Please resolve this issue if the submission is sufficient.

Reply:
>it would require revisiting https://github.com/spiffe/spire-plugin-sdk/pull/55 to not simply have a list of notes and a "valid" flag

I'm not sure that we really need to update the response message in the SDK.
The way I see it, the notes are pretty much a free form of adding additional information to the `valid` boolean. It may also include things like `Configuration foo is deprecated, use bar instead`. It may include many different things and indicating which refer to what may be difficult. Probably the best way to disambiguate that is to make sure that they are self-explanatory.

Reply:
Yes, the notes are supposed to be self explanatory.

My concern was only that if there were four notes, and two of them are errors, we can't sort out the two error notes from the two advisory notes.  We have no users of this, so it is a problem we haven't encountered (and maybe we never will encounter it).

At this time, I don't think it's too much for the person who ran the "spire-server validate" command to read all of the plugin's notes and figure out by context which ones are errors.

If you are happy with the output now, I am happy too, and we can close this one point in the review.